sync-core/src/test/java/com/cloudant/sync/query/AbstractQueryTestBase.java

```diff
@@ -17,27 +17,55 @@
 import static org.hamcrest.Matchers.containsInAnyOrder;
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
 
+import com.cloudant.sync.datastore.DatastoreExtended;
+import com.cloudant.sync.datastore.DatastoreManager;
 import com.cloudant.sync.datastore.DocumentBodyFactory;
 import com.cloudant.sync.datastore.DocumentRevision;
 import com.cloudant.sync.datastore.MutableDocumentRevision;
+import com.cloudant.sync.sqlite.SQLDatabase;
+import com.cloudant.sync.util.TestUtils;
 
+import org.junit.After;
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 
 import java.io.IOException;
 import java.sql.SQLException;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
-public class QueryExecutorTest extends AbstractIndexTestBase {
+/**
+ *  The aim is to make sure that the post hoc matcher class behaves the
+ *  same as the SQL query engine.
+ *
+ *  To accomplish this, we test the entire test execution pipeline contained
+ *  within this class using {@link com.cloudant.sync.query.QueryExecutorSQLOnlyTest}
+ *  which exercises the SQL query engine matching functionality.  We then test
+ *  the same test execution pipeline contained within this class using
+ *  {@link com.cloudant.sync.query.QueryExecutorMatcherTest} which in turn
+ *  exercises the post hoc matcher matching functionality.
+ */
+public abstract class AbstractQueryTestBase {
```

Comment:
Added a `QueryExecutorStandardTest` which uses the regular `IndexManager` and `QueryExecutor`. 
https://github.com/cloudant/sync-android/commit/cf527fd540554a34338857f179c44fa358444ad3#diff-ce1fd18195cc9f32262108611d0dd922R25
As of right now though only simple AND query is supported so only either the matcher or the SQL engine is being used to satisfy the results of a query.  In the next set of commits and PR I will be providing support for OR and subclauses so there will be a need at that point to add more tests so that the "standard" way of querying is validated appropriately through `QueryExecutorStandardTest`.